### DISCUSSION TITLE
XimInputMethod UpdateActive

### DISCUSSION THREAD

**Discussion by workgroupengineering** [AvaloniaUI/Avalonia#9990] [Q&A] [answered]
The variable active on the next line is always false because _controlActive is declared but never assigned.

https://github.com/AvaloniaUI/Avalonia/blob/7dd09cd2a5e85bd006265a5a4e6a2c39b087c4e8/src/Avalonia.X11/X11Window.Xim.cs#L73

@kekekeks   What is the correct behavior?

edit:
DBusTextInputMethodBase also has the same issue

https://github.com/AvaloniaUI/Avalonia/blob/7dd09cd2a5e85bd006265a5a4e6a2c39b087c4e8/src/Avalonia.FreeDesktop/DBusIme/DBusTextInputMethodBase.cs#L192

**Comment by kekekeks** [comment, marked as answer]
In the [original commit](https://github.com/AvaloniaUI/Avalonia/commit/35b365d97db414896d6aa56b402ea5003e9bbf8f#diff-9542d85d0ec6a4db6a7503fd237041f1636775c8c2bff8a9e4c986111d9cdf79) it was updated from `ITextInputMethodImpl::SetActive(bool active)`. 

In [this commit](https://github.com/AvaloniaUI/Avalonia/commit/5c37405259263c9f9a5f328ff9fa082357edbf5b#diff-fd001316af0ae91aebdaedda4515cbce54ef911b5aa00230cc63265579ceda44R208-R212) the field was replaced with `bool IsActive => _client != null` but it seems that not everything got updated.

**Reply by kekekeks** [reply]
The general idea was that IME should be active when window has input focus and an active text input client.
